src/stacks-hierarchy/Root_PurchaseOverviewScreen/use-offer-state.ts

```diff
@@ -55,16 +55,28 @@ const getValidDurationSeconds = (offer: Offer): number | undefined =>
     ? secondsBetween(offer.valid_from, offer.valid_to)
     : undefined;
 
+const getOfferForTraveller = (offers: Offer[], userTypeString: string) => {
+  const offersForTraveller = offers.filter(
+    (o) => o.traveller_id === userTypeString,
+  );
+
+  // If there are multiple offers for the same traveller, use the cheapest one.
+  // This shouldn't happen in practice, but it's a sensible fallback.
+  const offersSortedByPrice = offersForTraveller.sort(
```

Comment:
What if the offer I am looking for is the expensive one? It might not be part of this PR but would be nice to have a selector for complex cases.

Reply:
Shouldn't happen in practice, so it shouldn't matter. It's there just because it probably makes "more" sense to pick the cheap one if we get two offers by mistake :)